modules/widgets/src/compass-widget.tsx

```diff
@@ -0,0 +1,112 @@
+import {FlyToInterpolator, WebMercatorViewport, _GlobeViewport} from '@deck.gl/core';
+import type {Deck, Viewport, Widget, WidgetPlacement} from '@deck.gl/core';
+import {h, render} from 'preact';
+
+interface CompassWidgetProps {
+  id: string;
+  viewId?: string | null;
+  placement?: WidgetPlacement;
+  label?: string;
+  transitionDuration?: number;
+  style?: Partial<CSSStyleDeclaration>;
+  className?: string;
+}
+
+class CompassWidget implements Widget<CompassWidgetProps> {
+  id = 'compass';
+  props: CompassWidgetProps;
+  placement: WidgetPlacement = 'top-left';
+  viewId = null;
+  viewport: Viewport;
+  deck?: Deck;
+  element?: HTMLDivElement;
+
+  constructor(props: CompassWidgetProps) {
+    this.id = props.id || 'compass';
+    this.viewId = props.viewId || null;
+    this.placement = props.placement || 'top-left';
+    props.transitionDuration = props.transitionDuration || 200;
+    props.label = props.label || 'Compass';
+    props.style = props.style || {};
+    this.props = props;
+  }
+
+  setProps(props: CompassWidgetProps) {
+    Object.assign(this.props, props);
+  }
+
+  onViewportChange(viewport) {
+    this.viewport = viewport;
+    this.update();
+  }
+
+  onAdd({deck}: {deck: Deck}): HTMLDivElement {
+    const {style, className} = this.props;
+    const element = document.createElement('div');
+    element.classList.add('deckgl-widget', 'deckgl-widget-compass');
```

Comment:
After discussion with @ibgreen and @Pessimistress, we've decided to drop the `gl`. Couple reasons, it's shorter, we're long-term distancing from `gl` (onto WebGPU), and tooltip widget has always used `.deck-tooltip`.